src/vm/hints/hint_utils.rs

```diff
@@ -275,3 +275,68 @@ pub fn assert_le_felt(
         _ => Err(VirtualMachineError::FailedToGetIds),
     }
 }
+
+//Implements hint: from starkware.cairo.lang.vm.relocatable import RelocatableValue
+//        both_ints = isinstance(ids.a, int) and isinstance(ids.b, int)
+//        both_relocatable = (
+//            isinstance(ids.a, RelocatableValue) and isinstance(ids.b, RelocatableValue) and
+//            ids.a.segment_index == ids.b.segment_index)
+//        assert both_ints or both_relocatable, \
+//            f'assert_not_equal failed: non-comparable values: {ids.a}, {ids.b}.'
+//        assert (ids.a - ids.b) % PRIME != 0, f'assert_not_equal failed: {ids.a} = {ids.b}.'
+pub fn assert_not_equal(
+    vm: &mut VirtualMachine,
+    ids: HashMap<String, BigInt>,
+) -> Result<(), VirtualMachineError> {
+    //Check that ids contains the reference id for each variable used by the hint
+    let (a_ref, b_ref) = if let (Some(a_ref), Some(b_ref)) =
+        (ids.get(&String::from("a")), ids.get(&String::from("b")))
+    {
+        (a_ref, b_ref)
+    } else {
+        return Err(VirtualMachineError::IncorrectIds(
+            vec![String::from("a"), String::from("b")],
+            ids.into_keys().collect(),
+        ));
+    };
+    //Check that each reference id corresponds to a value in the reference manager
+    let (a_addr, b_addr) = if let (Some(a_addr), Some(b_addr)) = (
+        get_address_from_reference(a_ref, &vm.references, &vm.run_context),
+        get_address_from_reference(b_ref, &vm.references, &vm.run_context),
+    ) {
+        (a_addr, b_addr)
+    } else {
+        return Err(VirtualMachineError::FailedToGetIds);
+    };
+    //Check that the ids are in memory
+    match (vm.memory.get(&a_addr), vm.memory.get(&b_addr)) {
+        (Ok(Some(maybe_rel_a)), Ok(Some(maybe_rel_b))) => match (maybe_rel_a, maybe_rel_b) {
+            (MaybeRelocatable::Int(ref a), MaybeRelocatable::Int(ref b)) => {
+                if (a - b) % vm.prime.clone() == bigint!(0) {
```

Comment:
If `a` and `b` are in range this should only happen if `a - b == bigint!(0)`. Are those guaranteed to be in range? If that's so, omitting the division and `.clone()` should be more efficient.

Reply:
They are not always in range, so modulo is needed to compare them as field elements. % was replaced for mod_floor